### PR TITLE
Fix login and logout

### PR DESCRIPTION
**Pull request by zaucker**
- Add logout route
- Add logout method to `SessionUser`
- Cleanup frontend code a bit
- Misc cleanup

### CODE REVIEW

File: lib/Agrammon/Web/Routes.pm6 [hunk 1]
```diff
@@ -451,6 +451,14 @@ sub application-routes(Agrammon::Web::Service $ws) {
             content 'application/json', %cfg;
         }
 
+        post -> Agrammon::Web::SessionUser $user, 'logout' {
+            $user.logout();
+            content 'application/json', %(
+                :user($user.username),
+                :!sudoUser,
```
Comment (jnthn) [inline lib/Agrammon/Web/Routes.pm6:458]: Hm, why is this needed?

**Review by jnthn — APPROVED** [review]

File: lib/Agrammon/DB/User.pm6 [hunk 2]
```diff
@@ -100,7 +100,7 @@ class Agrammon::DB::User does Agrammon::DB {
         return self;
     }
 
-    method load {
+    method load() {
```
Comment (jnthn) [inline lib/Agrammon/DB/User.pm6:103]: I'm a little curious why this was needed, tbh...they should mean the same thing. What did you observe?

Comment (zaucker) [reply lib/Agrammon/DB/User.pm6:103]: > I'm a little curious why this was needed, tbh...they should mean the same thing. What did you observe?

Without `()` after `load` I got in `t/user.t`:

Screenshot 2021-02-08 at 15 19 02

File: lib/Agrammon/Web/Routes.pm6 (hunk 1 above)
Comment (zaucker) [reply lib/Agrammon/Web/Routes.pm6:458]: > Hm, why is this needed?

Do you mean the `:!sudoUser`? If so, that's something that is not implemented yet. The GUI has a `sudo` option for role `admin` so that an admin user can become another user without having to know his password (for supporting "help, my dataset doesn't work"). Upon `logout` the admin user is then returned to his own login account (and thus the GUI needs to know that it has to display the admin's username again and not open the login window.

Not sure if this is a bad idea from a security point of few. But I thought we could discuss this once I have implemented it.

Comment (jnthn) [reply lib/Agrammon/Web/Routes.pm6:458]: Yes, about `sudoUser`. The question was why that could ever matter at logout time; now I understand why!

File: lib/Agrammon/DB/User.pm6 (hunk 2 above)
Comment (jnthn) [reply lib/Agrammon/DB/User.pm6:103]: Hm, that sounds like a bug. We'll investigate that.
